editor/src/components/editor/use-default-collapsed-views.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import { useDispatch } from './store/dispatch-context'
+import { Substores, useEditorState } from './store/store-hook'
+import { isRight } from '../../core/shared/either'
+import { isJSXElement } from '../../core/shared/element-template'
+import * as EP from '../../core/shared/element-path'
+import { addCollapsedViews } from './actions/action-creators'
+
+export function useDefaultCollapsedViews() {
+  const dispatch = useDispatch()
+
+  // keep track of the previously collapsed element paths, or they will be re-collapsed
+  type PreviouslyCollapsed = { [path: string]: boolean }
+  const [previouslyCollapsed, setPreviouslyCollapsed] = React.useState<PreviouslyCollapsed>({})
```

Comment:
could this be a React.useRef? with the current value updated in the useEffect

Reply:
I guess it can! sec